manifests/agent/install.pp

```diff
@@ -1,8 +1,13 @@
 # Install the puppet client installation
 class puppet::agent::install {
-
-  package { $puppet::params::client_package:
-    ensure => $::puppet::version,
+  if ($::puppet::params::package_provider) {
+    package { $puppet::params::client_package:
+      ensure   => $::puppet::version,
+      provider => $::puppet::params::package_provider,
+    }
+  } else {
+    package { $puppet::params::client_package:
+      ensure => $::puppet::version,
+    }
   }
```

Comment:
I think you can even leave out the if. When it's set to undef, the OS default should be used.